public/app/features/dashboard-scene/sharing/ShareExportTab.tsx

```diff
@@ -43,46 +81,124 @@ export class ShareExportTab extends SceneObjectBase<ShareExportTabState> impleme
     });
   };
 
+  public onExportModeChange = (exportMode: ExportMode) => {
+    this.setState({
+      exportMode,
+    });
+
+    if (exportMode === ExportMode.Classic) {
+      this.setState({
+        isViewingYAML: false,
+      });
+    }
+  };
+
   public onViewJSON = () => {
     this.setState({
       isViewingJSON: !this.state.isViewingJSON,
     });
   };
 
+  public onViewYAML = () => {
+    this.setState({
+      isViewingYAML: !this.state.isViewingYAML,
+    });
+  };
+
   public getClipboardText() {
     return;
   }
 
   public getExportableDashboardJson = async (): Promise<{
-    json: Dashboard | DashboardJson | DashboardV2Spec | { error: unknown };
+    json: Dashboard | DashboardJson | DashboardV2Spec | ExportableResource | { error: unknown };
     hasLibraryPanels?: boolean;
+    initialSaveModelVersion: 'v1' | 'v2';
   }> => {
-    const { isSharingExternally } = this.state;
+    const { isSharingExternally, exportMode } = this.state;
 
     const scene = getDashboardSceneFor(this);
     const exportableDashboard = await scene.serializer.makeExportableExternally(scene);
+    const initialSaveModel = scene.getInitialSaveModel();
+    const initialSaveModelVersion = initialSaveModel && isDashboardV2Spec(initialSaveModel) ? 'v2' : 'v1';
     const origDashboard = scene.serializer.getSaveModel(scene);
     const exportable = isSharingExternally ? exportableDashboard : origDashboard;
+    const metadata = getMetadata(scene);
+
+    if (isDashboardV2Spec(origDashboard) && 'elements' in exportable) {
+      this.setState({
+        exportMode: ExportMode.V2Resource,
+      });
```

Comment:
@harisrozajac - this is incorrect. When viewing a v1 resource, the Toggle is set to V2 Resource. And, because of this `setState` that's called in async effect, it's impossible to switch to V1 Resource or Classic view - the `getExportableDashboardJson` overwrites asynchroniously whatever state the user picked to show.

Reply:
Here's the recording: 

https://github.com/user-attachments/assets/a00c7e54-c6bd-4f60-a0b7-231e3b07dcb5